src/main/java/wooteco/subway/domain/Sections.java

```diff
@@ -0,0 +1,145 @@
+package wooteco.subway.domain;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+
+public class Sections {
+
+    public final int limitSectionSize = 1;
+    private final List<Section> sections;
+
+    public Sections(List<Section> sections) {
+        this.sections = sections;
+    }
+
+    public boolean hasOneStation(Long stationId) {
+        long count = sections.stream()
+                .filter(it -> it.getUpStationId().equals(stationId) || it.getDownStationId().equals(stationId))
+                .count();
+        return count == 1;
+    }
+
+    public Long getSectionId(Long stationId) {
+        return sections.stream()
+                .filter(it -> it.getUpStationId().equals(stationId) || it.getDownStationId().equals(stationId))
+                .map(Section::getId)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("[ERROR] 삭제하려는 구간이 존재하지 않습니다."));
+    }
+
+    public Section getUpStationSection(Long stationId) {
+        return sections.stream()
+                .filter(it -> it.getDownStationId().equals(stationId))
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("[ERROR] 삭제할 대상의 상행종점을 찾을 수 없습니다."));
+    }
+
+    public Section getDownStationSection(Long stationId) {
+        return sections.stream()
+                .filter(it -> it.getUpStationId().equals(stationId))
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("[ERROR] 삭제할 대상의 하행종점을 찾을 수 없습니다."));
+    }
+
+    public void validateAddNewSection(Section newSection) {
+        validateSameStation(newSection.getUpStationId(), newSection.getDownStationId());
+        validateDistance(newSection.getDistance());
+        validateExistSection(newSection);
+    }
+
+    private void validateSameStation(Long upStationId, Long downStationId) {
+        if (upStationId.equals(downStationId)) {
+            throw new IllegalArgumentException("[ERROR] 동일한 지하철역을 입력할 수 없습니다.");
+        }
+    }
+
+    private boolean existSection(Long stationId) {
+        Long count = sections.stream()
+                .filter(it -> it.getUpStationId().equals(stationId) || it.getDownStationId().equals(stationId))
+                .count();
+        return count != 0;
+    }
+
+    private void validateExistSection(Section section) {
+        if (sections.size() == 0) {
+            return;
+        }
+        boolean hasSection1 = existSection(section.getUpStationId());
+        boolean hasSection2 = existSection(section.getDownStationId());
+
+        if (hasSection1 && hasSection2) {
+            throw new IllegalArgumentException("[ERROR] 이미 등록되어 있어 추가할 수 없습니다.");
+        }
+        if (hasSection1 == false && hasSection2 == false) {
```

Comment:
```suggestion
        if (!hasSection1 && !hasSection2) {
```
이런식으로 쓰지 않은 이유가 있나요?

Reply:
해당 부분도 좋은 가독성이 있었는데 미처 몰랐네요 😢 피드백 주신대로 변경하였습니다!

Reply:
사실 하나만 더 이야기 드리면 if문의 조건식은 부정보단 긍정으로 쓰는게 더 좋긴 하더라구요.
부정으로 쓰면 조금 헷갈림이 있는거 같아요 :)